packages/optimizer/lib/DomTransformer.js

```diff
@@ -168,9 +174,28 @@ class DomTransformer {
    * @return {string} - the transformed html string
    */
   async transformHtml(html, params) {
-    const tree = await treeParser.parse(html);
-    await this.transformTree(tree, params);
-    return treeParser.serialize(tree);
+    async function transform() {
+      const tree = await this.doProfile("parsing", () => treeParser.parse(html));
+
+      await this.doProfile('transform', () => this.transformTree(tree, params));
+
+      return treeParser.serialize(tree);
```

Comment:
should we also profile the serialization?

```suggestion
      return this.doProfile("serialization", () => treeParser.serialize(tree));
```

Reply:
Personally, I always prefer monitor things that don't need to be monitored until we need the metric.
I have added the metrics to this image because we were experiencing performance issues and we did not know where to search, so I needed to monitor anything

let me know

Reply:
my bad, I misread your comment, I totaly agree !